### ISSUE TITLE
Support building outside of the source root

### ISSUE THREAD

**Issue by plicease** [Perl5-Alien/Alien-Build#31]
Both autoconf and cmake support building outside of the source root, eg.:

```
../source-1.0.0/configre --prefix=...
```

and

```
cmake -Dblahblah=foo ../source-1.00/configure
```

In some cases it would save a lot of copying of files to build in an empty build root with the source root in a different directory.

**Comment by zmughal** [comment]
There are some projects that require this such as ITK which has [a check against building in source](https://github.com/InsightSoftwareConsortium/ITK/blob/master/CMake/PreventInSourceBuilds.cmake).

I may plan on making an Alien for this, but it is a rather large library.

**Comment by plicease** [comment]
You could still get something like ITK to work with some hooks around extract (and maybe build), but it would be nice to have a standard way to do this within AB core.

**Comment by plicease** [comment]
@zmughal any thoughts on nomenclature for this?  autoconf seems to refer to this as "VPATH build", cmake doesn't seem to have a name (after some brief googling).  I am inclined to not use the autoconf name, admittedly not for technical reasons.  But `vpath` is short and to the point.

**Comment by zmughal** [comment]
I concur with "out-of-source". That is more general and I've seen it also used with Qt as well (which has its own build system separate from GNU's or CMake's).

**Comment by plicease** [comment]
This is now supported with the `out_of_source` meta property.  The extract method extracts to a extract_XXXX directory instead of build_XXXX and build_XXXX is empty instead of containing the source.  The autoconf and cmake plugins do the right thing out of the box if the `out_of_source` property is set.  Things remaining undone:

 1. the test is failing on windows, I will (hopefully) fix this prior to a 1.08_01 release, and (definitely) before a production release.
 2. Alien::Build 1.08 should (automatically) be a prereq for any `alienfile` that uses the `out_of_source` meta property.

This feature is very close though.

**Comment by plicease** [comment]
This feature is in 1.08_01.  I plan on doing a production release as soon as we get a greenish looking matrix on a dev release.
